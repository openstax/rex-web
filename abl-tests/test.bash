#!/bin/bash

set -e

update_snapshot=$1
my_dir=$(dirname "$0")

if [[ $1 == '--help' ]]; then
    echo "This script runs a test on multiple input files (*.input.json)"
    echo "and compares each output to a corresponding snapshot file (*.snapshot.json)"
    echo ""
    echo "When the output differs from the snapshot the test fails"
    echo ""
    echo "--update-snapshots      Updates all snapshot files"
    exit 1
fi

shopt -s globstar nullglob
for input_file in "$my_dir"/*.input.json; do
    snapshot_file=${input_file%.input.json}.snapshot.json
    output_file=${input_file%.input.json}.temp.json
    if [[ $update_snapshot || ! -f $snapshot_file ]]; then
        output_file=$snapshot_file
    fi
    
    # -------------------------
    # Actual test goes here
    # -------------------------
    json=$(cat "$input_file")
    node "$my_dir/../script/entry.js" transform-approved-books-data --data "$json" > "$output_file"


    diff "$snapshot_file" "$output_file"
    
    if [[ "$snapshot_file" != "$output_file" ]]; then
        rm "$output_file"
    fi
done
shopt -u globstar nullglob