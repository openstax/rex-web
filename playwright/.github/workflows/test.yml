name: CI

on: [push, pull_request]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: 14
    - run: npm ci
    - run: npx playwright install-deps
    - run: npx playwright install
    - run: npm run ci-coverage
    - run: npm run test

    - name: Coveralls
      uses: coverallsapp/github-action@1.1.3
      with:
        github-token: ${{ secrets.GITHUB_TOKEN }}
