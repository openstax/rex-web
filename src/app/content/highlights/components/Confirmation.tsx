import { HTMLElement } from '@openstax/types/lib.dom';
import Color from 'color';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/macro';
import Button, { ButtonGroup } from '../../../components/Button';
import { labelStyle } from '../../../components/Typography';
import { useDrawFocus } from '../../../reactUtils';
import theme from '../../../theme';
import { mergeRefs } from '../../../utils';
import { cardPadding } from '../constants';
import { cardBorder } from './style';

// tslint:disable-next-line:variable-name
export const Overlay = styled.div`
  background: ${Color(theme.color.black).alpha(0.90).string()};
  outline: none;
  ${cardBorder}
  transition: background 200ms;
  position: absolute;
  display: flex;
  flex-direction: column;
  padding: 1.6rem;
  top: 0;
  right: 0;
  left: 0;
  overflow: visible;
  min-height: 100%;

  label {
    ${labelStyle}
    flex: 1;
    color: ${theme.color.text.white};
    margin-bottom: ${cardPadding}rem;
    overflow: visible;
  }
`;

interface Props {
  message: string;
  'data-analytics-region'?: string;
  'data-analytics-label'?: string;
  confirmMessage: string;
  confirmLink?: string;
  onConfirm?: () => void;
  onCancel: () => void;
  always?: () => void;
}

// tslint:disable-next-line:variable-name
const Confirmation = React.forwardRef<HTMLElement, Props>((
  {message, confirmMessage, confirmLink, always, onCancel, onConfirm, ...props}: Props,
  ref
) => {
  return <Overlay
    ref={mergeRefs(ref, useDrawFocus())}
    tabIndex={-1}
    {...props['data-analytics-region']
      ? {'data-analytics-region': props['data-analytics-region']}
      : {}
    }
  >
    <FormattedMessage id={message}>
      {(msg) => <label>{msg}</label>}
    </FormattedMessage>
    <ButtonGroup>
      <FormattedMessage id={confirmMessage}>
        {(msg) => <Button
          size='small'
          data-analytics-label={props['data-analytics-label'] ? props['data-analytics-label'] : 'confirm'}
          data-testid='confirm'
          variant='primary'
          onClick={(e: React.FormEvent) => {
            if (!confirmLink) {
              e.preventDefault();
            }
            if (onConfirm) {
              onConfirm();
            }
            if (always) {
              always();
            }
          }}
          {...confirmLink
            // eslint-disable-next-line
            ? {component: <a href={confirmLink} />}
            : {}
          }
        >{msg}</Button>}
      </FormattedMessage>
      <FormattedMessage id='i18n:highlighting:button:cancel'>
        {(msg) => <Button
          size='small'
          data-analytics-label='cancel'
          data-testid='cancel'
          onClick={(e: React.FormEvent) => {
            e.preventDefault();
            onCancel();
            if (always) {
              always();
            }
          }}
        >{msg}</Button>}
      </FormattedMessage>
    </ButtonGroup>
  </Overlay>;
});

export default Confirmation;
