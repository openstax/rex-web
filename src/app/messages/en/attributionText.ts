// tslint:disable max-line-length
export const defaultText = `
<p>
  Want to cite, share, or modify this book? This book uses the
  <a target="_blank" rel="noopener" href="{bookLicenseUrl}">
    {bookLicenseName} {bookLicenseVersion}
  </a> and you must attribute {copyrightHolder}.{originalMaterialLink, select,
    null {}
    other { The original material is available at: <a target="_blank" rel="noopener" href="{originalMaterialLink}">
      {originalMaterialLink}
    </a>.}
  }{copyrightHolder, select,
    OpenStax {}
    other { Changes were made to the original material, including updates to art, structure, and other content updates.}}
</p>

<strong>Attribution information</strong>
<ul>
  <li>
    If you are redistributing all or part of this book in a print format,
    then you must include on every physical page the following attribution:
    <blockquote>
      Access for free at https://openstax.org{introPageUrl}
    </blockquote>
  </li>
  <li>
    If you are redistributing all or part of this book in a digital format,
    then you must include on every digital page view the following attribution:
    <blockquote>
      Access for free at <a href="https://openstax.org{introPageUrl}">https://openstax.org{introPageUrl}</a>
    </blockquote>
  </li>
</ul>

<strong>Citation information</strong>
<ul>
  <li>
    Use the information below to generate a citation. We recommend using a
    citation tool such as
    <a target="_blank" rel="noopener" href="https://www.lib.ncsu.edu/citationbuilder/#/default/default">this one</a>.
    <ul>
      <li>
        Authors: {bookAuthors}
      </li>
      <li>
        Publisher/website: OpenStax
      </li>
      <li>
        Book title: {bookTitle}
      </li>
      <li>
        Publication date: {bookPublishDate, date, medium}
      </li>
      <li>
        Location: Houston, Texas
      </li>
      <li>
        Book URL: <a href="https://openstax.org{introPageUrl}">https://openstax.org{introPageUrl}</a>
      </li>
      <li>
        Section URL: <a href="https://openstax.org{currentPath}">https://openstax.org{currentPath}</a>
      </li>
    </ul>
  </li>
</ul>

<p>
  Â© {bookLatestRevision, date, medium} {copyrightHolder}. {copyrightHolder, select,
    OpenStax {Textbook content produced by OpenStax is licensed under a {bookLicenseName} {bookLicenseVersion}. }
    other {}}<strong>The OpenStax name, OpenStax logo, OpenStax book covers, OpenStax CNX name, and OpenStax CNX logo
  are not subject to the Creative Commons license and may not be reproduced without the prior and express written
  consent of Rice University.</strong>
</p>
`;
