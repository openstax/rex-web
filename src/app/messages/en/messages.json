{
  "i18n:a11y:skipToContent": "Skip to Content",
  "i18n:a11y:accessibilityHelp": "Go to accessibility page",
  "i18n:a11y:accessibilityHelp:link": "https://openstax.org/accessibility-statement",
  "i18n:attribution:toggle": "Citation/Attribution",
  "i18n:content:footnotes:title": "Footnotes",
  "i18n:content:solution:hide": "hide solution",
  "i18n:content:solution:show": "show solution",
  "i18n:content:solution:toggle-title": "Show/Hide Solution",
  "i18n:discard:body": "You have an unsaved note on this page.",
  "i18n:discard:button:cancel": "Cancel",
  "i18n:discard:button:discard": "Discard changes",
  "i18n:discard:heading": "Discard unsaved changes?",
  "i18n:error:boundary:action-btn-text": "OK",
  "i18n:error:boundary:heading": "Error",
  "i18n:error:boundary:sub-heading": "Uh-oh, there's been a glitch",
  "i18n:footer:column1:contact-us": "Contact Us",
  "i18n:footer:column1:faqs": "FAQ",
  "i18n:footer:column1:help": "Help",
  "i18n:footer:column1:support-center": "Support Center",
  "i18n:footer:column2:careers": "Careers",
  "i18n:footer:column2:newsletter": "Newsletter",
  "i18n:footer:column2:openstax": "OpenStax",
  "i18n:footer:column2:press": "Press",
  "i18n:footer:column3:accessibility": "Accessibility Statement",
  "i18n:footer:column3:license": "Licensing",
  "i18n:footer:column3:policies": "Policies",
  "i18n:footer:column3:privacy-policy": "Privacy Policy",
  "i18n:footer:column3:terms": "Terms of Use",
  "i18n:footer:heading": "Our mission is to improve educational access and learning for everyone.",
  "i18n:footer:social:fb:alt": "facebook",
  "i18n:footer:social:ig:alt": "instagram",
  "i18n:footer:social:in:alt": "linkedin",
  "i18n:footer:social:rice-logo:alt": "Rice University logo",
  "i18n:footer:social:tw:alt": "twitter",
  "i18n:highlighter:highlight:end": "End of {style, select, search {search result} other {{style}}} highlight",
  "i18n:highlighter:highlight:start": "Start {style, select, search {search result} other {{style}}} highlight",
  "i18n:highlighting:button:cancel": "Cancel",
  "i18n:highlighting:button:delete": "Delete",
  "i18n:highlighting:button:save": "Save",
  "i18n:highlighting:card:aria-label": "Enter note's text",
  "i18n:highlighting:card:placeholder": "Add a note...",
  "i18n:highlighting:card:show-more": "show more",
  "i18n:highlighting:change-color": "Apply {color} highlight",
  "i18n:highlighting:colors:blue": "blue",
  "i18n:highlighting:colors:green": "green",
  "i18n:highlighting:colors:pink": "pink",
  "i18n:highlighting:colors:purple": "purple",
  "i18n:highlighting:colors:yellow": "yellow",
  "i18n:highlighting:confirmation:delete-both": "Are you sure you want to delete this note and highlight?",
  "i18n:highlighting:confirmation:delete-highlight": "Are you sure you want to delete this highlight?",
  "i18n:highlighting:confirmation:delete-note": "Are you sure you want to delete this note?",
  "i18n:highlighting:dropdown:add-note": "Add note",
  "i18n:highlighting:dropdown:delete": "Delete",
  "i18n:highlighting:dropdown:edit": "Edit",
  "i18n:highlighting:dropdown:edit:aria-label": "Edit this highlight",
  "i18n:highlighting:dropdown:go-to-highlight": "Go to highlight",
  "i18n:highlighting:filters:all": "All",
  "i18n:highlighting:filters:chapters": "Chapter",
  "i18n:highlighting:filters:colors": "Color",
  "i18n:highlighting:filters:filter-by:aria-label": "Filter highlights by {filter}",
  "i18n:highlighting:filters:none": "None",
  "i18n:highlighting:filters:remove:chapter": "Remove breadcrumb for chapter {filterValue}",
  "i18n:highlighting:filters:remove:color": "Remove breadcrumb for color {filterValue}",
  "i18n:highlighting:login:link": "Log in",
  "i18n:highlighting:login:prompt": "Log in to highlight and take notes. Itâ€™s 100% free.",
  "i18n:metadata:description": "This free textbook is an OpenStax resource written to increase student access to high-quality, peer-reviewed learning materials.",
  "i18n:metadata:title": "Ch. {number} ",
  "i18n:nav:hello:text": "Hi {name}",
  "i18n:nav:login:text": "Log in",
  "i18n:nav:logo:alt": "OpenStax Logo",
  "i18n:nav:logout:text": "Log out",
  "i18n:nav:profile:text": "Account Profile",
  "i18n:notification:appmessage:dismiss": "OK",
  "i18n:notification:appmessage:header": "Notification",
  "i18n:notification:cookies:body": "This site uses cookies to optimize your experience on our platform. By continuing to use this site you agree to use our cookies. If you'd like more information or would like to opt out please see our <a href=\"/privacy-policy\" target=\"blank\">Privacy Policy</a>",
  "i18n:notification:cookies:header": "Privacy and cookies",
  "i18n:notification:cookies:ok": "Got it!",
  "i18n:notification:update:body": "Click below to use the most recent version of this software.",
  "i18n:notification:update:header": "Updates Available",
  "i18n:notification:update:reload": "Update",
  "i18n:notification:toast:search:highlight-not-found": "We're unable to find that search result. Please try another.",
  "i18n:notification:toast:highlights:create-failure": "The highlight could not be saved. Please check your connection and try again.",
  "i18n:notification:toast:highlights:delete-failure": "The highlight could not be deleted. Please check your connection and try again.",
  "i18n:notification:toast:highlights:highlight-not-found": "We're unable to find that highlight. Please try another.",
  "i18n:notification:toast:highlights:update-failure:color": "The highlight color could not be changed. Please check your connection and try again.",
  "i18n:notification:toast:highlights:update-failure:annotation": "The note could not be saved. Please check your connection and try again.",
  "i18n:notification:toast:highlights:load-failure": "We're unable to load Highlights on this page. Please check your connection and try again.",
  "i18n:notification:toast:highlights:popup:load-failure": "Some highlights could not be loaded. Please check your connection and try again.",
  "i18n:notification:toast:highlights:popup:print-failure": "We're unable to print this page. Please check your connection and try again.",
  "i18n:notification:toast:study-guides:load-failure": "We're unable to load Study Guides on this page. Please check your connection and try again.",
  "i18n:notification:toast:study-guides:popup:load-failure": "We're unable to load content on this page. Please check your connection and try again",
  "i18n:notification:toast:study-guides:popup:print-failure": "We're unable to print this page. Please check your connection and try again.",
  "i18n:nudge:study-tools:heading": "The study tools you need. <strong>100% FREE!</strong>",
  "i18n:nudge:study-tools:text:with-study-guides": "Boost your studies with expert-created study guides and highlighting. <strong>It's all free.</strong>",
  "i18n:nudge:study-tools:text:only-highlighting": "Highlight, take notes, and make your own study guides. <strong>It's all free.</strong>",
  "i18n:nudge:study-tools:aria-label:with-study-guides": "Image points to the Study Guides and My Highlights buttons in the toolbar. The study tools you need. 100% free. Boost your studies with expert-created study guides and highlighting. To close image, navigate to or click the x button.",
  "i18n:nudge:study-tools:aria-label:only-highlighting": "Image points to the My Highlights button in the toolbar. The study tools you need. 100% free. Highlight, take notes, and make your own study guides. To close image, navigate to or click the x button.",
  "i18n:page-not-found:heading": "Uh oh, we can't find the page you requested.",
  "i18n:page-not-found:text-before-button": "Try another page in the",
  "i18n:practice-questions:popup:close": "Click to close Practice Questions modal",
  "i18n:practice-questions:popup:continue": "Continue",
  "i18n:practice-questions:popup:correct": "Correct",
  "i18n:practice-questions:popup:empty:message": "There are no questions for this chapter/section.",
  "i18n:practice-questions:popup:empty:next-section": "Select a section from the dropdown above or continue to ",
  "i18n:practice-questions:popup:filters:chapters": "Chapter & Section",
  "i18n:practice-questions:popup:filters:filter-by:aria-label": "Filter PQ by {filter}",
  "i18n:practice-questions:popup:final": "You have reached the end of this set!",
  "i18n:practice-questions:popup:final:next-section": "Next section:",
  "i18n:practice-questions:popup:heading": "Practice Questions",
  "i18n:practice-questions:popup:incorrect": "Incorrect",
  "i18n:practice-questions:popup:intro:message": "{questions, plural, one {1 question in this section} other {{questions} questions in this section}}",
  "i18n:practice-questions:popup:intro:start": "Start now",
  "i18n:practice-questions:popup:navigation:finish": "Finish",
  "i18n:practice-questions:popup:navigation:next": "Next",
  "i18n:practice-questions:popup:navigation:next:after-submit-correct:aria-label": "This answer is correct. Next button.",
  "i18n:practice-questions:popup:navigation:show-answer": "Show answer",
  "i18n:practice-questions:popup:navigation:show-answer:after-submit-incorrect:aria-label": "This answer is incorrect. Show answer button.",
  "i18n:practice-questions:popup:navigation:skip": "Skip",
  "i18n:practice-questions:popup:navigation:submit": "Submit",
  "i18n:practice-questions:popup:questions": "Questions",
  "i18n:practice-questions:popup:read": "Read",
  "i18n:practice-questions:popup:warning-before-close": "Are you sure you want to exit this page? Your progress will not be saved.",
  "i18n:prevnext:next:aria-label": "Next Page",
  "i18n:prevnext:next:text": "Next",
  "i18n:prevnext:prev:aria-label": "Previous Page",
  "i18n:prevnext:prev:text": "Previous",
  "i18n:search-results:bar": "Search results sidebar",
  "i18n:search-results:bar:close-text:mobile": "Close",
  "i18n:search-results:bar:current-page": "Current Page",
  "i18n:search-results:bar:loading-state": "Loading results",
  "i18n:search-results:bar:query:no-results": "Sorry, no results found for ",
  "i18n:search-results:bar:query:results": "{total, plural, one {search result for } other {search results for }}",
  "i18n:search-results:bar:search-icon:value": "Search",
  "i18n:search-results:bar:toggle-text:mobile": "Search results",
  "i18n:studyguides:cta:button": "Sign Up",
  "i18n:studyguides:cta:info": "Like this guide? <br>We've got study guides for every chapter. <strong>Free forever.</strong>",
  "i18n:studyguides:cta:login": "Log in",
  "i18n:studyguides:cta:separator": "or",
  "i18n:studyguides:cta:title": "Expert-created study guides. <strong>100% free!</strong>",
  "i18n:studyguides:popup:filters:blue": "Important Concepts",
  "i18n:studyguides:popup:filters:green": "Connections & Relationships",
  "i18n:studyguides:popup:filters:purple": "Enrichment",
  "i18n:studyguides:popup:filters:yellow": "Key Terms",
  "i18n:studyguides:popup:filters:filter-by:aria-label": "Filter study guides by {filter}",
  "i18n:studyguides:popup:filters:remove:color": "Remove breadcrumb for label {filterValue, select, blue {Important Concepts} green {Connections & Relationships} purple {Enrichment} yellow {Key Terms}}",
  "i18n:studyguides:popup:filters:remove:chapter": "Remove breadcrumb for chapter {filterValue}",
  "i18n:studyguides:popup:no-highlights": "No results.",
  "i18n:studyguides:popup:no-highlights-tip": "Try selecting a different <strong>chapter or color filter</strong> to see different results.",
  "i18n:studyguides:popup:using-this-guide": "Using This Guide",
  "i18n:studyguides:popup:using-this-guide:alt": "The image presents the structure of study guides. The first element of each block summarizes, clarifies, or enriches key content. The second element is taken directly from the book. On the left side, there is a color label depending on the content.",
  "i18n:studyguides:popup:using-this-guide:close:aria-label": "Closes the image describing the structure of study guides.",
  "i18n:toc:title": "Table of Contents",
  "i18n:toc:toggle:closed": "Click to open the Table of Contents",
  "i18n:toc:toggle:opened": "Click to close the Table of Contents",
  "i18n:toolbar:buy-book:aria-label:text": "Buy book",
  "i18n:toolbar:buy-book:text": "Order a print copy",
  "i18n:practice-questions:popup:answers:choice": "Choice {choiceIndicator}",
  "i18n:toolbar:highlights:popup:body:add-highlight": "Make a highlight and add a note",
  "i18n:toolbar:highlights:popup:body:highlights-free": "Log in to highlight. 100% FREE.",
  "i18n:toolbar:highlights:popup:body:no-highlights-in-book": "You have no highlights in this book",
  "i18n:toolbar:highlights:popup:body:note:add-notes": "Add notes",
  "i18n:toolbar:highlights:popup:body:note:filter-chapters": "Filter by chapters & colors",
  "i18n:toolbar:highlights:popup:body:note:highlight": "Highlight",
  "i18n:toolbar:highlights:popup:body:note:review": "Review your highlights and notes in one place",
  "i18n:toolbar:highlights:popup:body:note:study-guide": "Make your study guide",
  "i18n:toolbar:highlights:popup:body:note:text": "Note:",
  "i18n:toolbar:highlights:popup:body:use-this-page": "Then use this page to create your own study guide",
  "i18n:toolbar:highlights:popup:button:back-to-top": "Go back to top",
  "i18n:toolbar:highlights:popup:close-button:aria-label": "Close My Highlights",
  "i18n:toolbar:highlights:popup:heading": "My Highlights and Notes",
  "i18n:toolbar:highlights:popup:heading:no-highlights": "No results.",
  "i18n:toolbar:highlights:popup:heading:no-highlights-tip": "Try selecting different <strong>chapter or color filters</strong> to see different results.",
  "i18n:toolbar:highlights:popup:help-info": "Need help? Visit the <a data-testid='support-link' data-analytics-label='support-link' target=\"_blank\" href=\"https://openstax.secure.force.com/help/articles/FAQ/Using-the-highlighting-and-note-taking-feature-on-mobile-or-tablet?search=highlighting\">Support Center</a>",
  "i18n:toolbar:highlights:text": "My highlights",
  "i18n:toolbar:practice-questions:button:text": "Practice",
  "i18n:toolbar:print:aria-label": "print",
  "i18n:toolbar:print:text": "Print",
  "i18n:toolbar:search:placeholder": "Search this book",
  "i18n:toolbar:search:toggle": "Search this book",
  "i18n:toolbar:studyguides:button:text": "Study guides",
  "i18n:toolbar:studyguides:popup:button:back-to-top": "Go back to top",
  "i18n:toolbar:studyguides:popup:close-button:aria-label": "Close Study Guides",
  "i18n:toolbar:studyguides:popup:heading": "Study Guide"
}
