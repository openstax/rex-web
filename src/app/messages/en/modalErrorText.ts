export const bodyWithLink = `
  We're not quite sure what went wrong. Restart your browser. If this doesn't solve
  the problem, visit our <a href="{supportCenterLink}" target="_blank">Support Center</a>.
`;
