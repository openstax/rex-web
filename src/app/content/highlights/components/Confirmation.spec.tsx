import React from 'react';
import { createIntl } from 'react-intl';
import renderer from 'react-test-renderer';
import en from '../../../../app/messages/en';
import createTestServices from '../../../../test/createTestServices';
import { makeFindByTestId } from '../../../../test/reactutils';
import TestContainer from '../../../../test/TestContainer';
import Confirmation from './Confirmation';

// this is a hack because useEffect is currently not called
// when using jsdom? https://github.com/facebook/react/issues/14050
// seems to work better in react-test-renderer but
// i need the ref here
jest.mock('react', () => {
  const react = (jest as any).requireActual('react');
  return { ...react, useEffect: react.useLayoutEffect };
});

describe('Confirmation', () => {

  const intl = createIntl({
    locale: 'en',
    messages: {
      ...en,
      confirm: 'confirm',
      message: 'message',
    },
  });
  const services = createTestServices();
  services.intl = intl;

  it('matches snapshot no selection', () => {
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        data-analytics-region='region'
        confirmMessage='confirm'
        onConfirm={() => null}
        onCancel={() => null}
      />
    </TestContainer>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('prevents default when clicking confirm button', () => {
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        confirmMessage='confirm'
        onCancel={() => null}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    const button = findByTestId('confirm');

    const preventDefault = jest.fn();
    button.props.onClick({preventDefault});

    expect(preventDefault).toHaveBeenCalled();
  });

  it('prevents default when clicking cancel button', () => {
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        confirmMessage='confirm'
        onCancel={() => null}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    const button = findByTestId('cancel');

    const preventDefault = jest.fn();
    button.props.onClick({preventDefault});

    expect(preventDefault).toHaveBeenCalled();
  });

  it('doesn\'t prevent default when clicking confirm link', () => {
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        confirmLink='/asdf'
        message='message'
        confirmMessage='confirm'
        onCancel={() => null}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    const button = findByTestId('confirm');

    const preventDefault = jest.fn();
    button.props.onClick({preventDefault});

    expect(preventDefault).not.toHaveBeenCalled();
  });

  it('calls onConfirm', () => {
    const onConfirm = jest.fn();
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        confirmMessage='confirm'
        onConfirm={onConfirm}
        onCancel={() => null}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    const button = findByTestId('confirm');
    button.props.onClick({preventDefault: jest.fn()});

    expect(onConfirm).toHaveBeenCalled();
  });

  it('calls onCancel', () => {
    const onCancel = jest.fn();
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        confirmMessage='confirm'
        onConfirm={() => null}
        onCancel={onCancel}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    const button = findByTestId('cancel');
    button.props.onClick({preventDefault: jest.fn()});

    expect(onCancel).toHaveBeenCalled();
  });

  it('calls always', () => {
    const always = jest.fn();
    const component = renderer.create(<TestContainer services={services}>
      <Confirmation
        message='message'
        confirmMessage='confirm'
        onCancel={() => null}
        always={always}
      />
    </TestContainer>);

    const findByTestId = makeFindByTestId(component.root);
    findByTestId('confirm').props.onClick({preventDefault: jest.fn()});
    findByTestId('cancel').props.onClick({preventDefault: jest.fn()});

    expect(always).toHaveBeenCalledTimes(2);
  });
});
