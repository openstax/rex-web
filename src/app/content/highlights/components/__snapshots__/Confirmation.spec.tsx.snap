// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation matches snapshot no selection 1`] = `
<div
  className="Confirmation__Overlay-stssi9-0 jzhfSA"
  data-analytics-region="region"
  tabIndex={-1}
>
  <label>
    message
  </label>
  <div
    className="Button__ButtonGroup-ayg7nk-1 fJPuTt"
  >
    <button
      className="Button-ayg7nk-0 dIvXiu"
      data-analytics-label="confirm"
      data-testid="confirm"
      onClick={[Function]}
    >
      confirm
    </button>
    <button
      className="Button-ayg7nk-0 bHTkEr"
      data-analytics-label="cancel"
      data-testid="cancel"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
